tensorboard/backend/event_processing/data_provider_test.py

```diff
@@ -211,6 +218,64 @@ def test_read_scalars_but_not_rank_0(self):
           run_tag_filter=run_tag_filter,
       )
 
+  def test_list_tensors_all(self):
+    provider = self.create_provider()
+    result = provider.list_tensors(
+        experiment_id="unused",
+        plugin_name=image_metadata.PLUGIN_NAME,
+        run_tag_filter=None,
+    )
+    self.assertItemsEqual(result.keys(), ["pictures"])
+    self.assertItemsEqual(result["pictures"].keys(), ["purple", "green"])
+    sample = result["pictures"]["purple"]
+    self.assertIsInstance(sample, base_provider.TensorTimeSeries)
+    self.assertEqual(sample.max_step, 10)
+    # nothing to test for wall time, as it can't be mocked out
+    self.assertEqual(sample.plugin_content, b"")
+    self.assertEqual(sample.display_name, "")  # not written by V2 summary ops
+    self.assertEqual(sample.description, "`#F0F`")
+
+  def test_list_tensors_filters(self):
+    provider = self.create_provider()
+
+    # Quick check only, as scalars and tensors use the same underlying
+    # filtering implementation.
+    result = provider.list_tensors(
+        experiment_id="unused",
+        plugin_name=image_metadata.PLUGIN_NAME,
+        run_tag_filter=base_provider.RunTagFilter(["pictures"], ["green"]),
+    )
+    self.assertItemsEqual(result.keys(), ["pictures"])
+    self.assertItemsEqual(result["pictures"].keys(), ["green"])
+
+  def test_read_tensors(self):
+    multiplexer = self.create_multiplexer()
+    provider = data_provider.MultiplexerDataProvider(multiplexer, self.logdir)
+
+    run_tag_filter = base_provider.RunTagFilter(
+        runs=["pictures"],
+        tags=["purple", "green"],
+    )
+    result = provider.read_tensors(
+        experiment_id="unused",
+        plugin_name=image_metadata.PLUGIN_NAME,
+        run_tag_filter=run_tag_filter,
+        downsample=None,  # not yet implemented
+    )
+
+    self.assertItemsEqual(result.keys(), ["pictures"])
+    self.assertItemsEqual(result["pictures"].keys(), ["purple", "green"])
+    for run in result:
```

Comment:
optional: It may be easier to assert by forming an expected map of `TensorTimeSeries` and just do `assertEqual` since we implemented `eq`.

Reply:
Indeed, but I wanted to use `np.testing.assert_equal` for better error
messages than “`giant_numpy_array_1 != giant_numpy_array_2`”.